### PR TITLE
fix(alerts): Fix display of invalid search key

### LINKED ISSUE #70666: Creating an alert using `release.stage: adopted` in the Sentry web UI throws a 500 error
### Environment

SaaS (https://sentry.io/)

### Steps to Reproduce

1. Navigate to Alerts > Create Alert > Crash free user rate
2. Select a project and environment
3. for filters, enter `release.stage:adopted` 
4. Set some thresholds below (I used 99.5 as critical, 99.7 as warning and 99.75 as recovery)
5. Name the alert
6. Click "Add alert"

### Expected Result

The alert should be created and visible in the Alert list.

### Actual Result

An error appears in lower right of the screen:
Screenshot 2024-05-10 at 9 38 57 AM

And in the console the message 
```
Failed to load resource: the server responded with a status of 400 ()
us.sentry.io/api/0/organizations/REDACTED/alert-rules/?duplicateRule=false&referrer=alert_stream&sessionId=REDACTED&wizardV3=true:1 

Failed to load resource: the server responded with a status of 500 ()
```

### Product Area

Alerts

### Link

_No response_

### DSN

_No response_

### Version

_No response_

┆Issue is synchronized with this [Jira Improvement](https://getsentry.atlassian.net/browse/FEEDBACK-2203) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by priscilawebdev**
#### Problem 1
Currently, when users navigate to **Alerts -> Crash Free User Rate -> Set Conditions**, they encounter invalid search keys, such as `release.stage`. Attempting to save the alert results in a request error due to the invalid query.

#### Problem 2
Currently, when users navigate to **Alerts -> Number of Errors -> Set Conditions**, they lack sufficient search query keys typically available for an error dataset (discovered under the hood).

#### Problem 1
This issue was identified as a recent regression. The code was reverted to resolve this problem.

#### Problem 2
We have added discover fields to the supported error tags, along with the `is` tag, to ensure users have the necessary search query keys.

The search keys available for errors in this PR are the following:

```
[
    "is",
    "id",
    "timestamp",
    "timestamp.to_hour",
    "timestamp.to_day",
    "culprit",
    "location",
    "message",
    "platform.name",
    "environment",
    "release",
    "dist",
    "title",
    "event.type",
    "transaction",
    "unreal.crash_type",
    "user",
    "user.id",
    "user.email",
    "user.username",
    "user.ip",
    "sdk.name",
    "sdk.version",
    "http.method",
    "http.referer",
    "http.status_code",
    "http.url",
    "os.build",
    "os.kernel_version",
    "device.name",
    "device.brand",
    "device.locale",
    "device.uuid",
    "device.arch",
    "device.family",
    "device.battery_level",
    "device.orientation",
    "device.screen_density",
    "device.screen_dpi",
    "device.screen_height_pixels",
    "device.screen_width_pixels",
    "device.simulator",
    "device.online",
    "device.charging",
    "device.class",
    "geo.country_code",
    "geo.region",
    "geo.city",
    "geo.subdivision",
    "error.type",
    "error.value",
    "error.mechanism",
    "error.handled",
    "error.unhandled",
    "error.received",
    "error.main_thread",
    "level",
    "stack.abs_path",
    "stack.filename",
    "stack.package",
    "stack.module",
    "stack.function",
    "stack.in_app",
    "stack.colno",
    "stack.lineno",
    "stack.stack_level",
    "app.in_foreground",
    "trace",
    "trace.span",
    "trace.parent_span",
    "total.count",
    "project",
    "issue",
    "user.display"
]
```

closes https://github.com/getsentry/sentry/issues/70666#issuecomment-2210706838

### CODE REVIEW

File: static/app/views/alerts/wizard/options.tsx [hunk 1]
```diff
@@ -273,6 +274,13 @@ const INDEXED_PERFORMANCE_ALERTS_OMITTED_TAGS = [
   ...Object.values(ReplayClickFieldKey),
 ];
 
+const ERROR_SUPPORTED_TAGS = [
```
Comment (priscilawebdev) [inline static/app/views/alerts/wizard/options.tsx:277]: @rachrwang could you please confirm that all the tags you were looking for are now listed with my changes?

**Review by shruthilayaj — DISMISSED** [review]

File: static/app/components/events/searchBar.tsx [hunk 2]
```diff
@@ -304,7 +295,7 @@ function SearchBar(props: SearchBarProps) {
               )
             : Object.assign({}, STATIC_FIELD_TAGS_WITHOUT_TRACING);
 
-    Object.assign(combinedTags, tagsWithKind, STATIC_SEMVER_TAGS, supportedTags);
+    Object.assign(combinedTags, tagsWithKind, STATIC_FIELD_TAGS, STATIC_SEMVER_TAGS);
```
Comment (shruthilayaj) [inline static/app/components/events/searchBar.tsx:298]: also, `STATIC_FIELD_TAGS` is already assigned to combinedTags, why do we to assign it again here?

Comment (priscilawebdev) [reply static/app/components/events/searchBar.tsx:298]: I just reverted the code to what it was before

File: static/app/components/events/searchBar.tsx [hunk 3]
```diff
@@ -277,12 +274,6 @@ function SearchBar(props: SearchBarProps) {
     const measurementsWithKind = getMeasurementTags(measurements, customMeasurements);
     const orgHasPerformanceView = organization.features.includes('performance-view');
 
-    // If it is not a metric alert search bar and supportedTags has a value, return supportedTags
-    // If it is a metric alert search bar, combine supportedTags with getTagList tags
-    if (metricAlert === false && supportedTags !== undefined) {
-      return supportedTags;
-    }
-
```
[the remark this answers is not in the document]
Comment (priscilawebdev) [reply static/app/components/events/searchBar.tsx:285]: it was added recently ~2 months ago and it caused regressions

File: static/app/components/events/searchBar.tsx (hunk 2 above)
Comment (priscilawebdev) [reply static/app/components/events/searchBar.tsx:298]: but it does make sense! let me update it

**Review by shruthilayaj — APPROVED** [review]

File: static/app/components/events/searchBarFields.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import omit from 'lodash/omit';
```
Comment (nhsiehgit) [inline static/app/components/events/searchBarFields.tsx:1]: If this file is exporting only constants, can we label it a `searchBarFieldConstants.tsx`?

we don't do `.ts` vs. `.tsx` files to distinguish btw component vs. helpers do we? 😅

**Review by nhsiehgit — APPROVED** [review]
non blocking nits

@rachrwang there's still the question on the tags if you want to sign off on those?
otherwise 👍

File: static/app/components/events/searchBar.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (priscilawebdev) [reply static/app/components/events/searchBar.tsx:285]: @nhsiehgit , thanks for your review 🙏. I could theoretically open two separate PRs, but the dataset for errors would lack search keys until the second PR is merged. That's why I prefer to address everything in this PR.

For more context: The [PR](https://github.com/getsentry/sentry/pull/69806) that regressed the search for the metrics dataset also improved the search query when the error dataset is selected.

File: static/app/components/events/searchBarFields.tsx (hunk 4 above)
Comment (priscilawebdev) [reply static/app/components/events/searchBarFields.tsx:1]: hey that is a good feedback - I believe we started to use only `.ts` when we do not use `jsx`. I will update and rename the file 🙂

File: static/app/views/alerts/wizard/options.tsx (hunk 1 above)
Comment (rachrwang) [reply static/app/views/alerts/wizard/options.tsx:277]: Tags look good to me, thanks Priscila!

**Pull request merged**